diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8da828efb..50e4bf6aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -383,7 +383,7 @@ add_subdirectory(externals)
 
 function(set_yuzu_qt_components)
     # Best practice is to ask for all components at once, so they are from the same version
-    set(YUZU_QT_COMPONENTS2 Core Widgets Concurrent)
+    set(YUZU_QT_COMPONENTS2 Core Gui Widgets Concurrent GuiPrivate)
     if (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
         list(APPEND YUZU_QT_COMPONENTS2 DBus)
     endif()
diff --git a/src/yuzu/CMakeLists.txt b/src/yuzu/CMakeLists.txt
index b67df498f..70c5edc5c 100644
--- a/src/yuzu/CMakeLists.txt
+++ b/src/yuzu/CMakeLists.txt
@@ -382,7 +382,7 @@ elseif(WIN32)
 endif()
 
 target_link_libraries(yuzu PRIVATE common core input_common frontend_common network video_core)
-target_link_libraries(yuzu PRIVATE Boost::headers glad Qt${QT_MAJOR_VERSION}::Widgets httplib::httplib)
+target_link_libraries(yuzu PRIVATE Boost::headers glad Qt${QT_MAJOR_VERSION}::Widgets Qt${QT_MAJOR_VERSION}::GuiPrivate httplib::httplib)
 target_link_libraries(yuzu PRIVATE ${PLATFORM_LIBRARIES} Threads::Threads)
 
 target_link_libraries(yuzu PRIVATE Vulkan::Headers)
